resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ResourceTypeHelper.scala

```diff
@@ -0,0 +1,180 @@
+    }
+
+    if (!ResourceTypeHelper.isYarnResourceTypesAvailable()) {
+      logWarning(resourceTypesNotAvailableErrorMessage)
+      return ""
+    }
+
+    var res: String = ""
+    try {
+      val resUtilsClass = Utils.classForName(
+        "org.apache.hadoop.yarn.util.resource.ResourceUtils")
+      val getNumberOfResourceTypesMethod = resUtilsClass.getMethod("getNumberOfKnownResourceTypes")
+      val numberOfResourceTypes: Int = getNumberOfResourceTypesMethod.invoke(null).asInstanceOf[Int]
+      val resourceClass = Utils.classForName(
+        "org.apache.hadoop.yarn.api.records.Resource")
+
+      // skip memory and vcores (index 0 and 1)
+      for (i <- 2 until numberOfResourceTypes) {
+        val getResourceInfoMethod = resourceClass.getMethod("getResourceInformation",
+          classOf[Int])
+        res ++= getResourceInfoMethod.invoke(resource, i.asInstanceOf[AnyRef]).toString()
+      }
+    } catch {
+      case e: InvocationTargetException =>
+        if (e.getCause != null) {
+          throw e.getCause
+        } else {
+          throw e
+        }
+      case NonFatal(e) =>
+        logWarning(resourceTypesNotAvailableErrorMessage, e)
+
+    }
+    res
+  }
+
+  def getAmountAndUnit(s: String): (Long, String) = {
+    try {
+      val AMOUNT_AND_UNIT_REGEX(amount, unit) = s
+      (amount.toLong, convertToYarnResourceFormat(unit))
+    } catch {
+      case _: MatchError => throw new IllegalArgumentException(
+        s"Value of resource type should match pattern $AMOUNT_AND_UNIT_REGEX, unmatched value: $s")
+    }
+  }
+
+  def convertToYarnResourceFormat(unit: String): String = {
+    unit match {
+      case "m" => "M"
+      case "g" => "G"
+      case "t" => "T"
+      case "p" => "P"
+      case _ => unit
+    }
+  }
+
+  private def createResourceInformation(
+      resourceName: String,
+      amount: Long,
+      unit: String,
+      resInfoClass: Class[_]) = {
+    val resourceInformation =
+      if (unit.nonEmpty) {
+        val resInfoNewInstanceMethod = resInfoClass.getMethod("newInstance",
+          classOf[String], classOf[String], classOf[Long])
+        resInfoNewInstanceMethod.invoke(null, resourceName, unit, amount.asInstanceOf[AnyRef])
+      } else {
+        val resInfoNewInstanceMethod = resInfoClass.getMethod("newInstance",
+          classOf[String], classOf[Long])
+        resInfoNewInstanceMethod.invoke(null, resourceName, amount.asInstanceOf[AnyRef])
+      }
+    resourceInformation
+  }
+
+  /**
+   * Checks whether Hadoop 2.x or 3 is used as a dependency.
+   * In case of Hadoop 3 and later,
+   * the ResourceInformation class should be available on the classpath.
+   */
+  def isYarnResourceTypesAvailable(): Boolean = {
+    try {
+      Utils.classForName("org.apache.hadoop.yarn.api.records.ResourceInformation")
+      true
+    } catch {
+      case NonFatal(e) =>
+        logWarning(resourceTypesNotAvailableErrorMessage, e)
```

Comment:
Remove. This will be noisy for the majority of people, who don't have (and are not even using) this feature.

Reply:
Removed all warning logs with this message.

Reply:
Removed all the warning logs